OWML.Launcher/SocketListener.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Newtonsoft.Json;
+using OWML.Common;
+using OWML.ModHelper;
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OWML.Launcher
+{
+    public class SocketListener
+    {
+        private const int BufferSize = 1024;
+        private static int _port;
+
+        public SocketListener(int port)
+        {
+            _port = port;
+            new Task(SetupSocketListener).Start();
+        }
+
+        private void SetupSocketListener()
+        {
+            TcpListener server = null;
+            try
+            {
+                ListenToSocket(ref server);
+            }
+            catch (SocketException ex)
+            {
+                ConsoleUtils.WriteByType(MessageType.Error, $"Error in socket listener: {ex}");
+            }
+            finally
+            {
+                server?.Stop();
+            }
+        }
+
+        private void ListenToSocket(ref TcpListener server)
+        {
+            var localAddress = IPAddress.Parse(Constants.LocalAddress);
```

Comment:
is the above comment about ref/out invisible? 🤔

Reply:
should be ok now?